server/map_rpc_server.go

```diff
@@ -384,8 +380,8 @@ func (t *TrillianMapServer) getWriteRevision(ctx context.Context, tree *trillian
 	if err != nil {
 		return 0, err
 	}
-	if assertRev != 0 && writeRev != assertRev {
-		return 0, status.Errorf(codes.FailedPrecondition, "can't write to revision %v", assertRev)
+	if writeRev != assertRev {
+		return 0, status.Errorf(codes.FailedPrecondition, "can't write to revision %v: only %d", assertRev, writeRev)
```

Comment:
Oups, this was added during the debugging phase, I'll restore the original line.

Reply:
Reminder to please address.